src/sdk/auth/browser-auth-provider.ts

```diff
@@ -37,17 +38,25 @@ export class IlandBrowserAuthProvider implements AuthProvider {
   async getToken(): Promise<string> {
     return new Promise<string>((resolve, reject) => {
       this._keycloak.updateToken(15).success(() => {
-        resolve(this._keycloak.token);
+        resolve(AuthProviderService.getImpersonateToken(this._keycloak.token));
       }).error(async() => {
         return this._init().then(() => {
-          resolve(this._keycloak.token);
+          resolve(AuthProviderService.getImpersonateToken(this._keycloak.token));
         }, (err) => {
           reject(err);
         });
       });
     });
   }
 
+  /**
+   * Return the current complete access token
+   * @returns {string} the access token
+   */
+  getTokenSync(): string {
+    return AuthProviderService.getImpersonateToken(this._keycloak.token);
```

Comment:
call this function from everywhere else internally so that we're only calling the augmentation function in one place

Reply:
i don't understand what you mean there. can you be more specific please ? :-/

Reply:
i mean call getTokenSync internally here vs. using getImpersonateToken all over the place

Reply:
ho !!!! ok i see :-)